fog/view/enclave/impl/src/lib.rs

```diff
@@ -431,6 +457,20 @@ fn get_block_data(mut responses: Vec<DecryptedMultiViewStoreQueryResponse>) -> B
     result
 }
 
+fn get_last_known_data(responses: &[DecryptedMultiViewStoreQueryResponse]) -> LastKnownData {
+    responses
+        .iter()
+        .max_by_key(|response| response.query_response.last_known_block_count)
+        .map_or(LastKnownData::default(), |response| {
```

Comment:
:wrench: micro optimization: one can consider that `map_or` will *always* call `LastKnownData::default()`. In this case it might get inlined. `map_or_else` will only call the function if the default is needed.
```suggestion
        .map_or_else(LastKnownData::default, |response| {
```

Reply:
Done.